search/elastic.py

```diff
@@ -23,252 +23,314 @@
+    """
+    filter_query_field = {"term": {field_name: field_value}}
     if isinstance(field_value, ValueRange):
         range_values = {}
         if field_value.lower:
-            range_values.update({"gte": field_value.lower_string})
+            range_values["gte"] = field_value.lower_string
         if field_value.upper:
-            range_values.update({"lte": field_value.upper_string})
-        filter_field = {
+            range_values["lte"] = field_value.upper_string
+        filter_query_field = {
             "range": {
                 field_name: range_values
             }
         }
     elif _is_iterable(field_value):
-        filter_field = {
+        filter_query_field = {
             "terms": {
                 field_name: field_value
-            }
+            },
         }
-    else:
-        filter_field = {
-            "term": {
-                field_name: field_value
-            }
-        }
-    return filter_field
+    return filter_query_field
 
 
 def _process_field_queries(field_dictionary):
     """
-    We have a field_dictionary - we want to match the values for an elasticsearch "match" query
-    This is only potentially useful when trying to tune certain search operations
+    Prepare ES query which mast be in the ES record set.
     """
-    def field_item(field):
-        """ format field match as "match" item for elasticsearch query """
-        return {
-            "match": {
-                field: field_dictionary[field]
-            }
-        }
-
-    return [field_item(field) for field in field_dictionary]
-
-
-def _process_field_filters(field_dictionary):
-    """
-    We have a field_dictionary - we match the values using a "term" filter in elasticsearch
-    """
-    return [_get_filter_field(field, field_value) for field, field_value in field_dictionary.items()]
+    return [
+        _get_filter_field(field, field_value)
+        for field, field_value in field_dictionary.items()
+    ]
 
 
 def _process_filters(filter_dictionary):
     """
-    We have a filter_dictionary - this means that if the field is included
-    and matches, then we can include, OR if the field is undefined, then we
-    assume it is safe to include
-    """
-    def filter_item(field):
-        """ format elasticsearch filter to pass if value matches OR field is not included """
-        if filter_dictionary[field] is not None:
-            return {
-                "or": [
-                    _get_filter_field(field, filter_dictionary[field]),
-                    {
-                        "missing": {
-                            "field": field
-                        }
-                    }
-                ]
-            }
+    Build list for filtering.
 
-        return {
-            "missing": {
-                "field": field
+    Match records where filtered fields may not exists.
+    """
+    filter_expression = []
+    for field in filter_dictionary:
+        if filter_dictionary[field]:
```

Comment:
Done. We don't checking None objects, because we add expression below which allows field can be empty or not exists ([ES exists doc](https://www.elastic.co/guide/en/elasticsearch/reference/7.8/query-dsl-exists-query.html#find-docs-null-values)). Default tests, doesn't check if filter field in the filter_dictionary is False